src/deps.ts

```diff
@@ -296,13 +296,36 @@ export interface AutoEncryptionOptions {
     /** Command line arguments to use when auto-spawning a mongocryptd */
     mongocryptdSpawnArgs?: string[];
     /**
-     * Full path to a CSFLE shared library to be used (instead of mongocryptd)
-     * @experimental
+     * Full path to a CSFLE shared library to be used (instead of mongocryptd).
+     *
+     * This needs to be the path to the file itself, not a directory.
+     * It can be an absolute or relative path. If the path is relative and
+     * its first component is `$ORIGIN`, it will be replaced by the directory
+     * containing the mongodb-client-encryption native addon file. Otherwise,
+     * the path will be interpreted relative to the current working directory.
+     *
+     * Currently, loading different CSFLE shared library files from different
+     * MongoClients in the same process is not supported.
+     *
+     * If this option is provided and no CSFLE shared library could be loaded
+     * from the specified location, creating the MongoClient will fail.
+     *
+     * If this option is not provided and `csfleRequired` is not specified,
+     * the AutoEncrypter will attempt to spawn and/or use mongocryptd according
+     * to the mongocryptd-specific `extraOptions` options.
      */
     csflePath?: string;
+    /**
+     * If specified, never use mongocryptd and instead fail when the CSFLE shared library
+     * could not be loaded.
+     *
+     * This is implied when `csflePath` is specified.
+     */
+    csfleRequired?: boolean;
     /**
      * Search paths for a CSFLE shared library to be used (instead of mongocryptd)
-     * @experimental
+     * Only for driver testing!
+     * @internal
      */
     csfleSearchPaths?: string[];
```

Comment:
Happy to drop this from the typings entirely if you prefer.

Reply:
Keeping it and marking it as internal means we could drop it any time, for now it's useful for testing so let's keep it :)